### ISSUE TITLE
PERF016: Detect same table accessed multiple times

### ISSUE THREAD

**Issue by RAprogramm** [RAprogramm/sql-query-analyzer#14]
## Summary

Add a rule to detect queries that reference the same table multiple times, which often indicates an opportunity for query optimization.

## Why This Matters

Multiple scans of the same table are usually avoidable:

- **Double I/O** — disk reads multiplied
- **Cache thrashing** — buffer pool efficiency drops
- **Query complexity** — often masks simpler solution
- **Maintenance burden** — changes must be made in multiple places

### Common Patterns

```sql
-- Anti-pattern: Multiple subqueries hitting same table
SELECT 
    (SELECT COUNT(*) FROM orders WHERE status = 'pending') as pending,
    (SELECT COUNT(*) FROM orders WHERE status = 'shipped') as shipped,
    (SELECT COUNT(*) FROM orders WHERE status = 'delivered') as delivered;

-- Better: Single scan with conditional aggregation
SELECT 
    COUNT(CASE WHEN status = 'pending' THEN 1 END) as pending,
    COUNT(CASE WHEN status = 'shipped' THEN 1 END) as shipped,
    COUNT(CASE WHEN status = 'delivered' THEN 1 END) as delivered
FROM orders;
```

## Rule Specification

| Property | Value |
|----------|-------|
| Rule ID | `PERF016` |
| Name | Multiple scans of same table |
| Category | Performance |
| Severity | Info |
| Default | Enabled |

### Detection Patterns

```sql
-- Should trigger
SELECT * FROM orders o1 
WHERE amount > (SELECT AVG(amount) FROM orders);

SELECT * FROM users 
WHERE id IN (SELECT user_id FROM orders) 
   OR id IN (SELECT user_id FROM orders WHERE status = 'vip');

-- Self-join (legitimate but worth flagging)
SELECT e.name, m.name as manager
FROM employees e
JOIN employees m ON e.manager_id = m.id;

-- Should NOT trigger (different tables)
SELECT * FROM orders o JOIN users u ON o.user_id = u.id;
```

### Suggested Message

> "Table 'orders' accessed multiple times in query. Consider combining into single scan using: (1) conditional aggregation (CASE WHEN), (2) window functions, or (3) CTE to materialize once."

## References

- [Use The Index, Luke - Combine Multiple Scans](https://use-the-index-luke.com/sql/partial-results/window-functions)
- [PostgreSQL Query Planning](https://www.postgresql.org/docs/current/using-explain.html)
- [MySQL Query Optimization](https://dev.mysql.com/doc/refman/8.0/en/optimization.html)

## Implementation Notes

- Count occurrences of each table name in query
- Trigger when same table appears 2+ times in FROM, JOIN, or subquery FROM
- Self-joins may be legitimate — use Info severity
- Suggest: conditional aggregation, window functions, CTE
- Parse aliases to identify same underlying table
